src/components/MessageLogger/LineMessage.tsx

```diff
@@ -1,15 +1,22 @@
 import React, { useEffect, useRef, useState } from "react";
 import styles from "@components/MessageLogger/LineMessage.module.scss";
-import { IoIosArrowDropdownCircle } from "react-icons/io";
 import { BsFillCaretRightFill } from "react-icons/bs";
 import { Message } from "@adapters/Message";
+import { Counter } from "@components/MessageLogger/Counter";
 
 interface Props {
   message: Message;
   count: number;
+  color: string;
+  colorBackground: string;
 }
 
-export const LineMessage = ({ message, count }: Props) => {
+export const LineMessage = ({
+  message,
+  count,
+  color,
+  colorBackground,
+}: Props) => {
   const [isOpen, setIsOpen] = useState(false);
   const [isOverflowing, setIsOverflowing] = useState(true);
   const [changeWidth, setChangeWidth] = useState(true);
```

Comment:
I dont understand the observer so its hard to know if there is an alternative, but you should use useState, as a notification for change. React changes always occour as a "reaction" to some meaningful change, not a explicit "change now" function.